src/float.rs

```diff
@@ -0,0 +1,28 @@
+/// A trait for extending [`prim@f32`] and [`prim@f64`] with extra methods.
+pub trait Float {
```

Comment:
It might be worth calling this `FloatEx` or something. Mostly because there is `num_traits::Float` already which provides the same interface as native `f32`/`f64` in Rust where as this is adding additional functionality. So I'm worried about clashing with the `num_traits` trait and also providing different functionality from what people might expect from a `Float` trait.

Reply:
I had the same concern but then promptly forgot about it, oops! I totally agree.

I've gone with `FloatExt` for the name as that's the suffix I've used when adding "extension" traits within Bevy. If that's alright.